python/mxnet/optimizer.py

```diff
@@ -80,6 +71,20 @@ def set_lr_scale(self, args_lrscale):
         """
         self.lr_scale = args_lrscale.copy()
 
+    def _update_count(self, index):
+        """
+        update num_update
+
+        Parameters:
+        index : int
+            The index will be updated
+        """
+        if index not in self._index_update_count:
+            self._index_update_count[index] = 0
+        self._index_update_count[index] += 1
+        self.num_update = max(self._index_update_count[index], self.num_update)
```

Comment:
all weight updating and the associated data, such as lr and momentum, are on the server nodes. a worker only read data, pull weight, compute gradient, and push gradient, nothing more. so a worker dose not have a learning rate on the distributed training.